src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi

```diff
@@ -57,7 +57,8 @@ cdef int _get_metadata(void *state,
       with nogil:
         cb(user_data, NULL, 0, status, c_error_details)
   args = context.service_url, context.method_name, callback,
-  _spawn_callback_async(<object>state, args)
+  plugin = <object>state
+  plugin._stored_ctx.run(_spawn_callback_async, plugin, args)
```

Comment:
This might break our Auth plugin tests, since most of them doesn't have `_stored_ctx` attribute?

Reply:
I don't think so. I've added `_stored_ctx` to `_Plugin`, which wraps the user-provided plugin. I also don't see such failures in the CI results, though I did just have to add support for runtimes that don't have `contextvars`.